examples/gwlb_with_vmseries/main.tf

```diff
@@ -0,0 +1,338 @@
+}
+
+# Manage the network required for the topology.
+module "vnet" {
+  source = "../../modules/vnet"
+
+  for_each = var.vnets
+
+  name                   = each.value.create_virtual_network ? "${var.name_prefix}${each.value.name}" : each.value.name
+  create_virtual_network = each.value.create_virtual_network
+  resource_group_name    = coalesce(each.value.resource_group_name, local.resource_group.name)
+  location               = var.location
+
+  address_space = each.value.address_space
+
+  create_subnets = each.value.create_subnets
+  subnets        = each.value.subnets
+
+  network_security_groups = { for k, v in each.value.network_security_groups : k => merge(v, { name = "${var.name_prefix}${v.name}" })
+  }
+  route_tables = { for k, v in each.value.route_tables : k => merge(v, { name = "${var.name_prefix}${v.name}" })
+  }
+
+  tags = var.tags
+}
+
+# create load balancers, both internal and external
+module "load_balancer" {
+  source = "../../modules/loadbalancer"
+
+  for_each = var.load_balancers
+
+  name                = "${var.name_prefix}${each.value.name}"
+  location            = var.location
+  resource_group_name = local.resource_group.name
+  zones               = each.value.zones
+
+  health_probes = each.value.health_probes
+
+  nsg_auto_rules_settings = try(
+    {
+      nsg_name = try(
+        "${var.name_prefix}${var.vnets[each.value.nsg_auto_rules_settings.nsg_vnet_key].network_security_groups[each.value.nsg_auto_rules_settings.nsg_key].name}",
+        each.value.nsg_auto_rules_settings.nsg_name
+      )
+      nsg_resource_group_name = try(
+        var.vnets[each.value.nsg_auto_rules_settings.nsg_vnet_key].resource_group_name,
+        each.value.nsg_auto_rules_settings.nsg_resource_group_name,
+        null
+      )
+      source_ips    = each.value.nsg_auto_rules_settings.source_ips
+      base_priority = each.value.nsg_auto_rules_settings.base_priority
+    },
+    null
+  )
+
+  frontend_ips = {
+    for k, v in each.value.frontend_ips : k => merge(
+      v,
+      {
+        public_ip_name = v.create_public_ip ? "${var.name_prefix}${v.public_ip_name}" : "${v.public_ip_name}",
+        subnet_id      = try(module.vnet[v.vnet_key].subnet_ids[v.subnet_key], null)
+        gwlb_fip_id    = try(module.gwlb[v.gwlb_key].frontend_ip_config_id, null)
+      }
+    )
+  }
+
+  tags       = var.tags
+  depends_on = [module.vnet]
+}
+
+# create Gateway Load Balancers
+module "gwlb" {
+  for_each = var.gateway_load_balancers
+  source   = "../../modules/gwlb"
+
+  name                = "${var.name_prefix}${each.value.name}"
+  resource_group_name = try(each.value.resource_group_name, local.resource_group.name)
+  location            = var.location
+
+  backends     = try(each.value.backends, null)
+  health_probe = try(each.value.health_probe, null)
+  lb_rule      = try(each.value.lb_rule, null)
+
+  zones = var.enable_zones ? try(each.value.zones, null) : null
```

Comment:
What does `each.value.zones` refer to if there's no `zones` property inside `gateway_load_balancers` variable?

Reply:
Thanks, I missed that. Add.

Reply:
I think that now variable `enable_zones` is redundant. You can eliminate it completely.

Reply:
Well, this I done by purpose - we have `enable_zones` for other examples. I wanted to have the same parts (as much as possible) in comparison to common example e.g. I removed NAT GW and application GW, but I added GWBL and test VMs. Should I remove `enable_zones` from all examples ?

Reply:
I remove `enable_zones` from GWLB example. In this PR I'm not touching other examples, ok ?

Reply:
Sure, thanks!